src/FailureDetectorConfig.md

```diff
@@ -85,3 +88,48 @@ config.setProperty("hazelcast.heartbeat.phiaccrual.failuredetector.sample.size",
 config.setProperty("hazelcast.heartbeat.phiaccrual.failuredetector.min.std.dev.millis", "100");
 [...]
 ```
+
+### Ping Failure Detector
+
+This failure detector is based on `InetAddress.isReachable()`.
+When the JVM process has enough permissions to create RAW sockets, the implementation will choose to rely on ICMP Echo requests, whereas, if not enough permissions, it will fallback on attempting a TCP Echo on port 7. In the latter, both a successful connection or an explicit rejection, will be treated as `Host is Reachable`.
+
+For the Ping Failure Detector to rely **only** on ICMP Echo requests, there are some criteria that need to be met.
+#### Requirements
+- Supported OS, as of Java 1.8 *nix environment support this
+- The Java executable must have the `cap_net_raw` capability. To do so, run `sudo setcap cap_net_raw=+ep path_to_java_exec`
+- When running with custom capabilities, the dynamic linker on Linux will reject loading libs from untrusted paths.
+To overcome, the `.../jre/lib/amd64/jli/` path needs to be added in the `ld.conf`
+`echo ".../jre/lib/amd64/jli/" >> /etc/ld.so.conf.d/java.conf && sudo ldconfig`
+- ICMP Echo Requests must not be blocked by the receiving hosts. `/proc/sys/net/ipv4/icmp_echo_ignore_all` set to `0`
+
+If any of the above criteria isn't met, then the `isReachable` will always fallback on TCP Echo attempts on port 7.
+
+To use the Ping Failure Detector.
+
+```xml
+<hazelcast>
+    [...]
+	<properties>
+        <property name="hazelcast.icmp.enabled">true</property>
+        <property name="hazelcast.icmp.parallel.mode">true</property>
+        <property name="hazelcast.icmp.timeout">500</property>
+        <property name="hazelcast.icmp.max.attempts">3</property>
+        <property name="hazelcast.icmp.interval">1000</property>
+        <property name="hazelcast.icmp.ttl">0</property>
+        [...]
+    </properties>
+    [...]
+</hazelcast>
+```
+
+- `hazelcast.icmp.timeout` Milliseconds until a ping attempt is considered failed if there was no reply.
+- `hazelcast.icmp.max.attempts` The maximum ping attempts before the member/node gets suspected by the detector.
+- `hazelcast.icmp.interval` The interval, in millis, between each ping attempt.
+- `hazelcast.icmp.ttl` The maximum number of hops the packets should go through or 0 for the default
+
+In the above configuration, the Ping detector will attempt 3 pings, one every second, and will wait up-to 500ms for each to complete. If after 3 seconds,
+there was no successful ping, the member will get suspected.
```

Comment:
I think you should also add default values. Either mention it on every property or say that the above example contains default values? I vote for option 1.